### ISSUE TITLE
Train Loss and Val Loss as nan

### ISSUE THREAD

**Issue by letdivedeep** [MaybeShewill-CV/lanenet-lane-detection#467]
I am training the model with the BDD-100K dataset for a single class with an input dim of 256 x 512. I have resized the images from BDD original dim of 720 x1280 to 256 x 512 before adding them to the TFrecords.

I am getting an error specific to train and val loss being nan. I have already removed images with no lane markings from the dataset. Still, I am getting the same error.

I ran an experiment for 10 epochs which results are attached below 
<img width="1680" alt="Screenshot 2020-11-10 at 11 46 48 AM" src="https://user-images.githubusercontent.com/74127861/98648823-cab8bc80-235c-11eb-85fe-d497fbf6a1a3.png">

**Comment by MaybeShewill-CV** [comment]
@letdivedeep Check whether the instance loss reach into nan. You need to adjust the instance loss params if so:)

**Comment by letdivedeep** [comment]
@MaybeShewill-CV  thanks for your reply

I found that the instance loss was ard 15.92 it didn't reach nan

**Comment by letdivedeep** [comment]
the eval result 
<img width="397" alt="Screenshot 2020-11-17 at 4 23 34 PM" src="https://user-images.githubusercontent.com/74127861/99381632-5e583300-28f1-11eb-9ad6-5df8ede8f75e.png">

**Comment by MaybeShewill-CV** [comment]
@letdivedeep instnace segmentation have learned something but seems not converged very well:)

**Comment by letdivedeep** [comment]
@MaybeShewill-CV thanks for your reply 

> @letdivedeep Check whether the instance loss reaches into nan. You need to adjust the instance loss params if so:)

> what do you mean by instance loss params and where to find it?

> and any specific values we need to set for bdd dataset?

I tried to change this config in the yaml and not the iloss is moving ard 6 not decreasing future :
```
POSTPROCESS:
    MIN_AREA_THRESHOLD: 150
    DBSCAN_EPS: 0.40
    DBSCAN_MIN_SAMPLES: 1500
```

**Comment by beiweixiaoxu** [comment]
> I am training the model with the BDD-100K dataset for a single class with an input dim of 256 x 512. I have resized the images from BDD original dim of 720 x1280 to 256 x 512 before adding them to the TFrecords.
> 
> I am getting an error specific to train and val loss being nan. I have already removed images with no lane markings from the dataset. Still, I am getting the same error.
> 
> I ran an experiment for 10 epochs which results are attached below
> <img alt="Screenshot 2020-11-10 at 11 46 48 AM" width="1680" src="https://user-images.githubusercontent.com/74127861/98648823-cab8bc80-235c-11eb-85fe-d497fbf6a1a3.png">

may i ask you two questions?
1. how do you know whether one image has lane markings or not?
2. images in bdd100k/drivable_maps/labels are "binary segmentation label" mentioned in the readme of this project, right?  however, i find all images in that path are pure black...

**Comment by MaybeShewill-CV** [comment]
@letdivedeep What you changed before had nothing relative to instance segmentation. The params was used in postprocess cluster procedure. Check this https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/66d7405d8e5284767addf31bab6a7ba17274d187/lanenet_model/lanenet_back_end.py#L157-L162

**Comment by MaybeShewill-CV** [comment]
@beiweixiaoxu 1. Check https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/66d7405d8e5284767addf31bab6a7ba17274d187/tools/test_lanenet.py#L125-L130 The mask will be empty if no lane exists
2.You may compare the BDD100K dataset with Tusimple dataset to find the difference:)

**Comment by beiweixiaoxu** [comment]
> @beiweixiaoxu 1. Check
> 
> https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/66d7405d8e5284767addf31bab6a7ba17274d187/tools/test_lanenet.py#L125-L130
> 
> The mask will be empty if no lane exists
> 2.You may compare the BDD100K dataset with Tusimple dataset to find the difference:)
Thanks for replying, i'm training on Tusimple now, then i'll train on BDD100K. Thanks again.
![image](https://user-images.githubusercontent.com/59757561/99866904-1b97a300-2bf0-11eb-9951-35017f9763d9.png)

**Comment by MaybeShewill-CV** [comment]
@beiweixiaoxu Welcome:)

**Comment by letdivedeep** [comment]
> @beiweixiaoxu 1. Check
> 
> https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/66d7405d8e5284767addf31bab6a7ba17274d187/tools/test_lanenet.py#L125-L130
> 
> The mask will be empty if no lane exists
> 2.You may compare the BDD100K dataset with Tusimple dataset to find the difference:)

thanks for the guidances the iloss started decreasing. i did the following changes 
```
            instance_segmentation_loss, l_var, l_dist, l_reg = \
                    lanenet_discriminative_loss.discriminative_loss(
                        pix_embedding, instance_label, self._embedding_dims,
                        pix_image_shape, 0.3, 2.0, 1.0, 1.0, 0.001
                    )
 
<img width="1680" alt="Screenshot 2020-11-23 at 11 37 22 AM" src="https://user-images.githubusercontent.com/74127861/99933275-8c6dc500-2d80-11eb-95ac-b69a80116cb9.png">

```

**Comment by letdivedeep** [comment]
> @letdivedeep What you changed before had nothing relative to instance segmentation. The params was used in postprocess cluster procedure. Check this
> 
> https://github.com/MaybeShewill-CV/lanenet-lane-detection/blob/66d7405d8e5284767addf31bab6a7ba17274d187/lanenet_model/lanenet_back_end.py#L157-L162

thanks for the guidances the iloss started decreasing. I did the following changes
        ```
    instance_segmentation_loss, l_var, l_dist, l_reg = \
                    lanenet_discriminative_loss.discriminative_loss(
                        pix_embedding, instance_label, self._embedding_dims,
                        pix_image_shape, 0.3, 2.0, 1.0, 1.0, 0.001
                    )
<img width="1680" alt="Screenshot 2020-11-23 at 11 37 22 AM" src="https://user-images.githubusercontent.com/74127861/99933338-c6d76200-2d80-11eb-9ba6-fac9d6647e13.png">


```

**Comment by MaybeShewill-CV** [comment]
@letdivedeep First you need to make sure which loss reach to nan. It's binary loss or instance seg loss. If it's the instance loss which reach to nan then you need to adjust the params above again:)

**Comment by letdivedeep** [comment]
> @letdivedeep First you need to make sure which loss reach to nan. It's binary loss or instance seg loss. If it's the instance loss which reach to nan then you need to adjust the params above again:)

@MaybeShewill-CV ..Sure will keep track of this... Thanks once again

**Comment by MaybeShewill-CV** [comment]
@letdivedeep Welcome:)

**Comment by letdivedeep** [comment]
@MaybeShewill-CV .. I used the above config of instance segmentation params

I trained the model with this param for 550 epochs. where my binary loss is ard 0.23 and iloss is ard 3.4. 

the iloss was moving ard 3.2 to 4 from 100 epochs itself but it's not going below that.

any point what may be wrong  ?

**Comment by MaybeShewill-CV** [comment]
@letdivedeep Can not point out which params is wrong precisely. Sry:)

**Comment by letdivedeep** [comment]
@MaybeShewill-CV, not a concern. Thanks, I will notify if found anything on this

**Comment by MaybeShewill-CV** [comment]
@letdivedeep Thanks for sharing:)
